src/wasm/wasm-type.cpp

```diff
@@ -533,6 +541,11 @@ Type Type::getLeastUpperBound(Type a, Type b) {
   if (a.size() != b.size()) {
     return Type::none; // a poison value that must not be consumed
   }
+  if (a.isRef()) {
+    // FIXME: `anyref` is only valid here if the `anyref` feature is enabled,
+    // but this information is not available within `Type` alone.
+    return b.isRef() ? Type::anyref : Type::none;
```

Comment:
Fixing this looks like it requires changing the signature of this function to `getLeastUpperBound(Type a, Type b, FeatureSet features)`, with call sites aware of enabled features passing these in.

Only works this way currently because we are not generating, for example, an `(if $condition $funcref $externref)` anywhere, but an API user can do so. Do you prefer to tackle this issue in this PR?

Reply:
Unless I'm missing some use of this function, I believe it will only incorrectly return `anyref` for invalid code. As long as validation will separately catch that invalidity, I'm tempted to not do anything with this issue besides document it. WDYT?

Reply:
Ah, it looks like this PR doesn't make any changes to validation yet. It would be good to fix that.

Reply:
Have a feeling that this will resurface eventually, but since the fuzzer isn't complaining I guess it is fine for now. Is the comment ok, or would you prefer a little more detail?

Reply:
Oh, going to check!

Reply:
On a first glimpse it looks like the validator uses `Type::isSubType` for its checks. This then leads to a similar issue like this one for `getLeastUpperBound`, but for `isSubType` not knowing the enabled features I guess, where the check in `isSubType` is `right == Type::anyref` which can only happen in valid code if `anyref` is enabled.

Can you point me into the right direction what else needs to be updated in the validator?

Reply:
On the other hand, there will be a(nother) validation error anyway if `anyref` is used but not enabled, via `Type::getFeatures()`.

Reply:
If this ends up generating `anyref` in other programs that don't use or enable `anyref`, eventually it will be caught by the validator, and that basically means one of Binaryen passes have a bug. So I think it's OK to keep this simple and not include types for now, but erroring out here might help Binaryen authors debug their code a little, or something.

I think `isSubType` will be OK, because it doesn't create `anyref` when it isn't enabled.

Reply:
> On the other hand, there will be a(nother) validation error anyway if `anyref` is used but not enabled, via `Type::getFeatures()`.

Aha, I had forgotten that this wouldn't require any changes in the validator itself. I agree with @aheejin that this should be fine, then. We can probably replace the `FIXME` with a comment pointing out the issue but explaining that it will only happen with invalid code.